doc/getting-started/gsg_quick_setup.sh

```diff
@@ -0,0 +1,277 @@
+    mkdir -p /mnt/EFI/acrn
+    echo "Copy $acrn_conf_path /mnt/loader/entries/"
+    if [[ ! -f $acrn_conf_path ]]; then
+        echo "Missing acrn.conf file in folder: $acrn_conf_path"
+        umount /mnt && sync
+        exit 1
+    fi
+    cp -r $acrn_conf_path /mnt/loader/entries/
+    if [[ $? -ne 0 ]]; then echo "You didn't choose an right EFI partition." && exit 1; fi
+    echo "Copy $acrn_efi_path to /mnt/EFI/acrn"
+    if [[ ! -f $acrn_efi_path ]]; then
+        echo "Missing acrn.efi file in folder: $acrn_efi_path"
+        umount /mnt && sync
+        exit 1
+    fi
+    cp -r $acrn_efi_path /mnt/EFI/acrn/
+    echo "Check ACRN efi boot event"
+    check_acrn_bootefi=`efibootmgr | grep ACRN`
+    if [[ "$check_arcn_bootefi" -ge "ACRN" ]]; then
+        echo "Clean all ACRN efi boot event"
+        efibootmgr | grep ACRN | cut -d'*' -f1 | cut -d't' -f2 | xargs -i efibootmgr -b {} -B
+    fi
+    echo "Check linux bootloader event"
+    number=$(expr `efibootmgr | grep 'Linux bootloader' | wc -l` - 1)
+    if [[ $number -ge 1 ]]; then
+        echo "Clean all Linux bootloader event"
+        efibootmgr | grep 'Linux bootloader' | cut -d'*' -f1 | cut -d't' -f2 | head -n$number | xargs -i efibootmgr -b {} -B
+    fi
+
+    echo "Add new ACRN efi boot event"
+    efibootmgr -c -l "\EFI\acrn\acrn.efi" -d $partition -p 1 -L "ACRN"
+
+    echo "Create loader.conf"
+    mv /mnt/loader/loader.conf /mnt/loader/loader.conf.bck && touch /mnt/loader/loader.conf
+    echo "Add default boot wait time"
+    echo "timeout 5" > /mnt/loader/loader.conf
+    echo "Add default boot to ACRN"
+    echo "default acrn" >> /mnt/loader/loader.conf
+
+    new_kernel=`ls /mnt/EFI/org.clearlinux/*sos* -tl | grep kernel | head -n1 | awk -F'/' '{print $5}'`
+    echo "Getting latest Service OS kernel version: $new_kernel"
+    cur_kernel=`cat /mnt/loader/entries/acrn.conf | sed -n 2p | cut -d'/' -f4`
+    echo "Getting current Service OS kernel version: $cur_kernel"
+
+    echo "Replacing root partition uuid in acrn.conf"
+    sed -i "s/<UUID of rootfs partition>/`blkid -s PARTUUID -o value $root_partition`/g" /mnt/loader/entries/acrn.conf
+
+    if [[ "$cur_kernel" == "$new_kernel" ]]; then
+        echo "No need to replace kernel conf info"
+    else
+        echo "Replace with new sos kernel in acrn.conf"
+        sed -i "s/$cur_kernel/$new_kernel/g" /mnt/loader/entries/acrn.conf
+    fi
+
+    echo "Service OS setup done!"
+    echo "Rebooting Service OS to taking effect changes."
+else
+    echo -e "Fail to upgrade sos $target_clearlinux from mirror url."
+fi
+
+umount /mnt
+sync
+reboot
+}
+
+function upgrade_uos()
+{
+# UOS download link
+uos_image_link="https://download.clearlinux.org/releases/$target_version/clear/clear-$target_version-kvm.img.xz"
+
+# Set proxy if needed.
+if [[ -n $proxy ]]; then
+    echo "Setting proxy to: $proxy"
+    export https_proxy=$proxy
+fi
+
+if [[ ! -z `findmnt /mnt -n` ]]; then umount /mnt; fi
```

Comment:
I think it's a bit rude of the script to silently unmount `/mnt` if it is already mounted. Perhaps the user is using this for some other purpose. It would be better to exit gracefully and ask him to free it up himself.

Reply:
Yes, it's fixed.